src/overlays/actors/ovl_Boss_01/z_boss_01.c

```diff
@@ -246,191 +647,2934 @@ ActorInit Boss_01_InitVars = {
+            }
+
+            Matrix_MultVec3f(&pos, &startVec);
+
+            startVec.x *= 0.2f;
+            startVec.y *= 0.2f;
+
+            baseX = (u16)(s32)(startVec.x + 32.0f);
+            baseY = (u16)((s32)startVec.y * 64);
+
+            if (sShadowSizes[i] == ODOLWA_SHADOW_SIZE_EXTRA_LARGE) {
+                for (y = 0, addY = -0x180; y < ARRAY_COUNT(sShadowExtraLargeMap); y++, addY += 0x40) {
+                    for (x = -sShadowExtraLargeMap[y]; x < sShadowExtraLargeMap[y]; x++) {
+                        index = baseX + x + baseY + addY;
+                        if ((index >= 0) && (index < ODOLWA_SHADOW_TEX_SIZE)) {
+                            tex[index] = 255;
+                        }
+                    }
+                }
+            } else if (sShadowSizes[i] == ODOLWA_SHADOW_SIZE_LARGE) {
+                for (y = 0, addY = -0x100; y < ARRAY_COUNT(sShadowLargeMap); y++, addY += 0x40) {
+                    for (x = -sShadowLargeMap[y]; x < sShadowLargeMap[y]; x++) {
+                        index = baseX + x + baseY + addY;
+                        if ((index >= 0) && (index < ODOLWA_SHADOW_TEX_SIZE)) {
+                            tex[index] = 255;
+                        }
+                    }
+                }
+            } else if (sShadowSizes[i] == ODOLWA_SHADOW_SIZE_MEDIUM) {
+                for (y = 0, addY = -0xC0; y < ARRAY_COUNT(sShadowMediumMap); y++, addY += 0x40) {
+                    for (x = -sShadowMediumMap[y]; x < sShadowMediumMap[y] - 1; x++) {
+                        index = baseX + x + baseY + addY;
+                        if ((index >= 0) && (index < ODOLWA_SHADOW_TEX_SIZE)) {
+                            tex[index] = 255;
+                        }
+                    }
+                }
+            } else {
+                for (y = 0, addY = -0x80; y < ARRAY_COUNT(sShadowSmallMap); y++, addY += 0x40) {
+                    for (x = -sShadowSmallMap[y]; x < sShadowSmallMap[y] - 1; x++) {
+                        index = baseX + x + baseY + addY;
+                        if ((index >= 0) && (index < ODOLWA_SHADOW_TEX_SIZE)) {
+                            tex[index] = 255;
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+void Boss01_GenShadowTex(u8* tex, Boss01* this, PlayState* play) {
+    s32* iter = (s32*)tex;
+    s16 i;
+
+    for (i = 0; i < (s32)(ODOLWA_SHADOW_TEX_SIZE / sizeof(s32)); i++, iter++) {
+        *iter = 0;
+    }
+
+    Matrix_RotateXFNew(1.0f);
+
+    for (i = 0; i <= 5; i++) {
+        Boss01_FillShadowTex(this, tex, i / 5.0f);
+    }
+}
+
+/**
+ * Draws Odolwa's dynamic shadow underneath him.
+ */
+void Boss01_DrawShadowTex(u8* tex, Boss01* this, PlayState* play) {
+    s32 pad[2];
+    f32 alpha;
+    GraphicsContext* gfxCtx = play->state.gfxCtx;
+
+    OPEN_DISPS(gfxCtx);
+
+    Gfx_SetupDL25_Opa(play->state.gfxCtx);
+
+    alpha = (400.0f - this->actor.world.pos.y) * 0.0025f;
```

Comment:
Does this match? Not sure if it is better
```suggestion
    alpha = (400.0f - this->actor.world.pos.y) / 400.0f;
```

Reply:
That would be cool if it matched, but it doesn't. However, this does, so I went with this instead:
```c
alpha = (400.0f - this->actor.world.pos.y) * (1.0f / 400.0f);
```